pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -30,23 +30,42 @@
   isPy3k, pythonOlder }:
 
 let
-  inherit (cudaPackages) cudatoolkit cudnn nccl;
+  inherit (cudaPackages) cudnn nccl;
 in
 
 # assert that everything needed for cuda is present and that the correct cuda versions are used
-assert !cudaSupport || (let majorIs = lib.versions.major cudatoolkit.version;
+assert cudaSupport -> (let majorIs = cudaPackages.cudaMajorVersion;
                         in majorIs == "9" || majorIs == "10" || majorIs == "11");
 
 # confirm that cudatoolkits are sync'd across dependencies
-assert !(MPISupport && cudaSupport) || mpi.cudatoolkit == cudatoolkit;
-assert !cudaSupport || magma.cudatoolkit == cudatoolkit;
+# TODO: verify it's OK to equality-compare cudaPackages (which are attrsets, scopes, or whatever)
```

Comment:
Yeah, that's a really interesting question. This was a nice ability when dealing with just one derivation. I have no idea how this works on package sets. Maybe @FRidh knows?

Reply:
My assumption was that Nix would shallowly compare the expressions, sort of as symbols. That'd be somewhat restrictive but tolerable: it'd essentially mean that users shouldn't mix cuda/cudnn instances coming from different evaluations of nixpkgs, but should write an overlay and evaluate a single consistent nixpkgs world